### PR TITLE
You are now your character when spawned into vallhalla

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

- This PR makes it so vallhalla copies your prefs when you spawn in, in other words you are your character in valhalla

## Why It's Good For The Game

I think it's nicer to be your character rather than some random gen with pink hair and a cyan beard. 

## Changelog
:cl:
qol: You are now your character when spawned into vallhalla
/:cl:

### CODE REVIEW

**Comment by Serxule** [comment]
wont this prevent you from spawning in normally as your character though? im pretty sure its why you get randomized

**Comment by FosterGrime-Heymynameslouie** [comment]
blessed pr, spede merhg



> wont this prevent you from spawning in normally as your character though? im pretty sure its why you get randomized

I believe this only happens if you go to valhalla first in the start of the round and go to respawn 30 minutes later resulting in the "there is already a character with this name!" error. 
I think

**Comment by Losenis** [comment]
Would need to be tested because in valhalla if you die or SSD your character is fully deleted, there's no corpse left or anything to actually hold the same name unlike normal respawn where you'd be able to find your old corpse if you went looking for it.

**Comment by ivanmixo** [comment]
Depends on how the system looks for the same name... which I can't test because my PC bricked itself this morning for no reason.

**Review by BraveMole — CHANGES_REQUESTED** [review]
This will conflict if you already spawned normally

**Comment by ivanmixo** [comment]
> This will conflict if you already spawned normally

Okay, I've tested this and no, it won't. The mobs spawned in valhalla never get their name added to the list of players joined (only the people who join the game through the lobby do), so they never conflict.

The call stack is basically `LateSpawn()` (on `/datum/game_mode`), `create_character()`  (on `/mob/new_player`), `on_spawn()` (on `/mob`, or `/mob/living/carbon/human`, in this case) which finally calls  `update_names_joined_list()` (on `/mob`), which adds the name to `GLOB.real_names_joined`, which is the list that gets checked for if the character name was already used in the round. 

TL;DR: It doesn't conflict, merge pleas.
